test/remote/gateways/remote_elavon_test.rb

```diff
@@ -205,8 +206,17 @@ def test_successful_purchase_with_custom_fields
     assert response.authorization
   end
 
-  def test_successful_purchase_with_currency
-    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(currency: 'JPY'))
+  def test_failed_purchase_with_multi_currency_terminal_setting_disabled
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(currency: 'USD', multi_currency: true))
+
+    assert_failure response
+    assert response.test?
+    assert_equal 'Transaction currency is not allowed for this terminal.  Your terminal must be setup with Multi currency', response.message
+    assert response.authorization
+  end
+
+  def test_successful_purchase_with_multi_currency
+    assert response = @multi_currency_gateway.purchase(@amount, @credit_card, @options.merge(currency: 'JPY', multi_currency: true))
```

Comment:
Do we want to test the gateway-level and option-level flags separately?

Reply:
I think that's a holdover from before I changed to use a gateway-level flag. I'll add a new test.